### ISSUE TITLE
using vue.js with select2

### ISSUE THREAD

**Issue by abhijitghogre** [vuejs/Discussion#281]
I am trying to use select2 plugin in vue using custom-directive approach.

I am facing a small issue of infinite looping. How do I fix this?

https://jsfiddle.net/9Lwo9r8y/

**Comment by jakeryansmith** [comment]
It looks like the update watcher triggers the change event which would in turn trigger the update watcher? So this would go on forever.

**Comment by abhijitghogre** [comment]
Yeah, any idea how to fix this?

Or is there any better searchable select plugin to use with vue.js?

**Comment by abhijitghogre** [comment]
@yyx990803 ?

**Comment by jakeryansmith** [comment]
I've been messing with this for the last few min, this is what I have so far: https://jsfiddle.net/9Lwo9r8y/

**Comment by abhijitghogre** [comment]
@jakeryansmith I think you pasted the same fiddle.

**Comment by jakeryansmith** [comment]
Oh sorry. What I had wasn't working that great anyway. Does this have to be a directive? I think the problem is that you set up an onChange listener on the select menu, but also have an updated listener with the directive. I'm not sure both are needed. You might try writing it in a way that you don't use both, idk just an idea. Or maybe don't even use a directive and just fill the select menu using ```<select option="college_classes" v-model="selected_college_class_id">```
Then change the value of selected_college_class_id when the select value changes.

**Comment by jakeryansmith** [comment]
This seems to work:
```
Vue.directive('select2', {
    bind: function () {
        var vm = this.vm;
        var key = this.expression;

        var select = $(this.el);

        select.select2();
        select.on('change', function () {
            vm.$set(key, select.val());
        });
    }
});

new Vue({
    el: '#main',
    data: {
        selected_college_class_id: '',
        college_classes: [{
            id: 1,
            name: 'FE'
        }, {
            id: 2,
            name: 'SE'
        }]
    }
});
```

I just removed the update listener.

**Comment by abhijitghogre** [comment]
Nope. If you remove update listener, select 2 will not select the appropriate option if the value of model changes. Try selecting the second option and click the 'set class_id as 1' button. It will change the model value but won't update the option.

**Comment by abhijitghogre** [comment]
I did this. It works but it is lot of code and not that clean way of doing it.

https://jsfiddle.net/9Lwo9r8y/3/

I hope @yyx990803 suggests the correct way to achieve this!

**Comment by jakeryansmith** [comment]
I don't see anything wrong with the changes you made. It actually looks a lot cleaner to me. Using the 'options' directive is the recommended way of filling a select menu. What exactly don't you like about it?

**Comment by abhijitghogre** [comment]
The fact that I have to manage 2 way data binding for select2 myself.
Imagine for a big app, I will need to register so many watchers and so many change event listeners.

**Comment by jakeryansmith** [comment]
Gotcha

**Comment by kaorun343** [comment]
Hello.

I checked the first code, and I found that `newVal` and `oldVal` were in reverse order.

**Comment by MacArthurJustin** [comment]
I know this is a bit old but I recently had to do this and found a bit easier way of connecting everything.
https://jsfiddle.net/reyekdnj/

**Comment by desprit** [comment]
@MacArthurJustin 
Your fiddle seems to be broken.

**Comment by MacArthurJustin** [comment]
@desprit Thanks I don't knwo why the ready option is there I must have not noticed when I saved it. try this one https://jsfiddle.net/reyekdnj/2/

**Comment by desprit** [comment]
@MacArthurJustin 
Thank you!

**Comment by ilgala** [comment]
I've created a little fiddle for Vuejs 2, Any ideas why the button isn't working?
https://jsfiddle.net/ilgala/82ztLqvL/
